File: 2021/02/02_dive.jl
lines = readlines("inputs/02")

function p1()
    pos = [0, 0]
    for line in lines
        dir, cnt = split(line, " ")
        if dir[1] == 'f'
            pos[1] += parse(Int, cnt)
        elseif dir[1] == 'd'
            pos[2] += parse(Int, cnt)
        else
            pos[2] -= parse(Int, cnt)
        end
    end
    println(prod(pos))
end
function p2()
    pos = [0, 0]
    aim = 0
    for line in lines
        dir, cnt = split(line, " ")
        cnt = parse(Int, cnt)
        if dir[1] == 'f'
            pos[1] += cnt
            pos[2] += aim * cnt
        elseif dir[1] == 'd'
            aim += cnt
        else
            aim -= cnt
        end
    end
    println(prod(pos))
end

p1()
p2()

File: 2021/02/02_dive.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <sstream>


using namespace std;

int main() {
	ifstream inp("inputs/02");

	string dir;
	int cnt;
	int pos[4] = {0, 0, 0, 0};
	int aim = 0;
	while (inp >> dir >> cnt){
		if (dir[0] == 'f'){
			pos[0] += cnt;
			pos[2] += cnt;
			pos[3] += aim * cnt;
		} else if (dir[0] == 'd'){
			pos[1] += cnt;
			aim += cnt;
		} else {
			pos[1] -= cnt;
			aim -= cnt;
		}
	}
	cout << pos[0] * pos[1] << endl;
	cout << pos[2] * pos[3] << endl;

	return 0;
}